app/main.py

```diff
@@ -1,8 +1,38 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name, age, wife=None, husband=None):
+        self.name = name
+        self.age = age
+
+        if husband is not None:
+            self.husband = husband
+        if wife is not None:
+            self.wife = wife
+
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+    result_list = []
+
+    for person in people:
+        name = person['name']
+        age = person['age']
+        if 'wife' in person:
+            wife = person['wife']
+            result_list.append(Person(name, age, wife=wife))
+        if 'husband' in person:
+            husband = person['husband']
+            result_list.append(Person(name, age, husband=husband))
```

Comment:
You added "wife" and "husband" twice. Here and at $32 - $36
Also we dont need $9 - $12, because we add with Person.people[person['name']].husband which automatly added self.husband or self.wife